### ISSUE TITLE
Advanced_Visualizer_Scenarios: How to create a VSIX installer for it this Debugger Visualiser?

### ISSUE THREAD

**Issue by PolarGoose** [Microsoft/VSSDK-Extensibility-Samples#296]
The [Advanced_Visualizer_Scenarios](https://github.com/microsoft/VSSDK-Extensibility-Samples/tree/master/Advanced_Visualizer_Scenarios/EvaluationTimeoutSample) example shows how to create a Debugger Visualizer.

The Readme file says that the plugin DLL needs to be copied to the `C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Packages\Debugger\Visualizers` folder.

Following this example, I managed to create my own Debugger Visualiser.
However, I don't know how to package it into a VSIX installer so that I can publish it on the https://marketplace.visualstudio.com/vs

The problem is that VSIX deploys the binaries to the folder like `C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VSExtensions\rybgvyx5.mow`
I can't find a way how to tell the installer to deploy my dll to the `C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Packages\Debugger\Visualizers`

Could you please help me?

**Comment by PolarGoose** [comment]
I have found another example that shows a workaround:
[VSX-Samples.VSIXDebuggerVisualizer](https://github.com/visualstudioextensibility/VSX-Samples/tree/master/VSIXDebuggerVisualizer)
How it works:
* Implementation: [VSIX/VSIXPackage.cs Initialize() method](https://github.com/visualstudioextensibility/VSX-Samples/blob/master/VSIXDebuggerVisualizer/VSIX/VSIXPackage.cs#L23)
* They create a dummy VS extension that, when loaded, copies the `DebuggerVisualizer.dll` to the `My Documents\<VisualStudioVersion>\Visualizers` folder

The problem is that when the extension is removed, the `DebuggerVisualizer.dll` remains installed.

**Comment by rezanid** [comment]
The only option I can think of is to create a Windows Installer package (MSI). Using an MSI installer you can add files to different locations and they will be removed when your package is uninstalled. You can read more here:
https://learn.microsoft.com/en-us/visualstudio/extensibility/preparing-extensions-for-windows-installer-deployment?view=vs-2022

I wouldn't down that road if I can avoid it though. AFAIK Windows Installer extensions are not supported in VS Marketplace.

**Comment by PolarGoose** [comment]
@rezanid,

Thank you.
Yes, indeed, it is unfortunate.
I ended up using a new VS Extensibility API that allows the creation of a VSIX package for the debugger visualizer:
https://github.com/PolarGoose/VisualStudio_ExceptionGraphVisualizer_plugin
However, the API is very limited. Especially the GUI part.

**Comment by rezanid** [comment]
Indeed, but you did a good job there. I am also working on an extension with many moving parts. You might find useful bits there if you take time. It's not super clean though 😁
Https://github.com/rezanid/xrmtools
